fxgcppubsub/reactor/ack/reactor.go

```diff
@@ -0,0 +1,34 @@
+package ack
+
+import (
+	"cloud.google.com/go/pubsub/apiv1/pubsubpb"
+	"github.com/ankorstore/yokai-contrib/fxgcppubsub/reactor"
+)
+
+// AckReactor is a pub/sub test server reactor for subscriptions message acks.
+type AckReactor struct {
+	supervisor reactor.WaiterSupervisor
+}
+
+// NewAckReactor returns a new AckReactor instance.
+func NewAckReactor(supervisor reactor.WaiterSupervisor) *AckReactor {
+	return &AckReactor{
+		supervisor: supervisor,
+	}
+}
+
+// FuncNames returns the list of function names this reactor will react to.
+func (r *AckReactor) FuncNames() []string {
+	return []string{
+		"Acknowledge",
+	}
+}
+
+// React is the reactor logic.
+func (r *AckReactor) React(req interface{}) (handled bool, ret interface{}, err error) {
```

Comment:
error returned is always nil?

Reply:
yes, it's a reactor chain, I just hook here to notify the waiters supervisor to stop, and let the chain continue, where the actual server will decide if there is an error or not